docs/grammar/SolidityParser.g4

```diff
@@ -175,6 +175,17 @@ locals[
 	 )*
 	(Returns LParen returnParameters=parameterList RParen)?
 	(Semicolon | body=block);
+
+/**
+ * The definition of a free function.
+ */
+ freeFunctionDefinition:
+ 	Function identifier
```

Comment:
`fallback` and `receive` seem to be allowed for free functions (they only generate a warning).

```suggestion
 	Function (identifier | Fallback | Receive)
```

So we probably don't have any test cases covering this? Please add some if we don't.

Reply:
Oh, I just assumed that they would not be allowed. I am going to add the tests.